modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java

```diff
@@ -1563,6 +1606,34 @@ private BaseModel<ExternalReferenceCodeModel> _getManyToOneRelatedModel(
 			objectRelationship.getObjectRelationshipId(), primaryKey);
 	}
 
+	private String _getNewValue(
+			long groupId, ObjectDefinition objectDefinition, String value)
+		throws Exception {
+
+		String copy = _language.get(LocaleUtil.getSiteDefault(), "copy");
+
+		String newValue = StringBundler.concat(
+			value, StringPool.SPACE, StringPool.OPEN_PARENTHESIS, copy,
+			StringPool.CLOSE_PARENTHESIS);
+
+		String prefix = value;
+
+		for (int i = 1;; i++) {
+			long count = objectEntryLocalService.getValuesListCount(
+				new Long[] {groupId}, objectDefinition.getCompanyId(),
+				objectDefinition.getUserId(),
+				objectDefinition.getObjectDefinitionId(), null, newValue);
```

Comment:
I didn't quite understand your last comment. Did you try passing the Predicate to use the filter only by the title field column? Something like [this](https://github.com/liferay/liferay-portal/blob/2aec8d8de73cd96fb681cc276382badd2c2e76da/modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryLocalServiceTest.java#L4698-L4715).
If you use the search feature, it will search all fields in the object. This wouldn't be good for performance.